Packs/ArmisEventCollector/Integrations/ArmisEventCollector/ArmisEventCollector.py

```diff
@@ -0,0 +1,427 @@
+
+    2.  Most recent event has later timestamp then other events in the response.
+        Meaning: This is the normal case where events in the response have different timestamps.
+        Handle: Return list of dedup event and new list of 'new_ids' for next run.
+
+    3.  Empty event list (no new events received from API response).
+        Meaning: Usually means there are not any more events to fetch at the moment.
+        Handle: Return empty list and the unchanged list of 'events_last_fetch_ids' for next run.
+
+    Args:
+        events (list[dict]): List of events from the current fetch response.
+        events_last_fetch_ids (list[str]): List of IDs of events from last fetch cycle.
+        unique_id_key (str): Unique event ID key of specific event type (Alert, Threat Activity etc.)
+
+    Returns:
+        tuple[list[dict], list[str]: The list of dedup events and ID list of events of current fetch.
+    """
+    # remove events that already appeared in last fetch
+    dedup_events: list[dict] = [event for event in events if event.get(unique_id_key) not in events_last_fetch_ids]
+    # get ids of dedup events
+    new_ids: list[str] = [event.get(unique_id_key, '') for event in dedup_events]
+
+    if not events:
+        return dedup_events, events_last_fetch_ids
+
+    latest_event_from_response = arg_to_datetime(events[-1].get('time'))
+    earliest_event_from_response = arg_to_datetime(events[0].get('time'))
+
+    if isinstance(latest_event_from_response, datetime) and isinstance(earliest_event_from_response, datetime)\
+            and are_two_event_time_equal(latest_event_from_response, earliest_event_from_response):
+        events_last_fetch_ids.extend(new_ids)
+        return dedup_events, events_last_fetch_ids
+    else:
+        return dedup_events, new_ids
+
+
+def fetch_by_event_type(event_type: dict, events: list, next_run: dict, client: Client,
+                        max_fetch: int, last_run: dict, fetch_start_time_param: datetime):
+    """ Fetch events by specific event type.
+
+    Args:
+        event_type (dict): Dictionary containing event type unique ID key, AQL query and type name.
+        events (list): List of fetched events.
+        next_run (dict): Last run dictionary for next fetch cycle.
+        client (Client): Armis client to use for API calls.
+        max_fetch (int): Max number of events to fetch.
+        last_run (dict): Last run dictionary.
+        fetch_start_time_param (datetime): Fetch start time (usually current time).
+    """
+    unique_id_key, aql_query, type = event_type['unique_id_key'], event_type['aql_query'], event_type['type']
+    last_fetch_ids = f'{type}_last_fetch_ids'
+    last_fetch_time = f'{type}_last_fetch_time'
+
+    demisto.debug(f'debug-log: handling event-type: {type}')
+    if last_run:
+        event_type_fetch_start_time = calculate_fetch_start_time(last_run.get(last_fetch_time), fetch_start_time_param)
```

Comment:
when running get command with "from_date" = "1 day" but last fetch was 3 minutes ago, this will run on 3 minutes? When running the get-command we still have the last run.

Reply:
Fixed from_date parameter handling.